### PR TITLE
feat(blob): add azure blob storage provider

### PR DESCRIPTION
**Pull request by Mossaka**
This commit adds a azblob storage implementation for the blobstore interface as a provider. 

I have not tested this out yet but I believe the core functionality is in this PR. Will make this ready to review after I have tested it out and make sure everything works.

### CODE REVIEW

File: crates/providers/blobstore-azblob/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,529 @@
...
+        match builder.await {
+            Ok(_) => PutObjectResponse { stream_id: None },
+            Err(e) => {
+                error!(error = %e, "failed to put object");
+                PutObjectResponse { stream_id: None }
+            }
+        }
+    }
+
+    async fn get_object(&self, ctx: Context, req: GetObjectRequest) -> GetObjectResponse {
```
Comment (Mossaka) [inline crates/providers/blobstore-azblob/src/lib.rs:457]: Need to implement trunk reads... Otherwise this will be super slow

**Review by brooksmtownsend — COMMENTED** [review]
I only had a few comments, the logic of this provider looks great! When it comes to testing this provider, I'd love to help out getting a declarative app manifest together for testing 😄

File: crates/providers/blobstore-azblob/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,529 @@
...
+                error!("failed to retrieve client: {e}");
+                return Vec::new();
+            }
+        };
+
+        let container_client = client.container_client(&arg.container_id);
+        let mut results = Vec::new();
+        for obj in arg.objects.iter() {
+            let blob_client = container_client.blob_client(obj);
+            match blob_client.delete().into_future().await {
```
Comment (Mossaka) [inline crates/providers/blobstore-azblob/src/lib.rs:390]: It doesn't seem like az-blob-rust-sdk provides a API for deleting multiple blobs in one req.

File: crates/providers/blobstore-azblob/src/config.rs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        }
+
+        if let Ok(access_key) = env::var("STORAGE_ACCESS_KEY") {
+            config.storage_access_key = access_key;
+        }
+
+        Ok(config)
+    }
+
+    pub fn configure_az(self) -> StorageCredentials {
```
Comment (ahmedtadde) [inline crates/providers/blobstore-azblob/src/config.rs:57]: wouldn't this consume the object?... is this the intended outcome? ignore this comment if `StorageConfig => StorageCredentials` is a desired transform.

**Comment by brooksmtownsend** [comment]
@Mossaka this is awesome! As long as it's ok with you, I'm going to take a look at integrating the new provider SDK into this to make it easy on you 😄

**Comment by brooksmtownsend** [comment]
Closing in favor of #1694, all work and commits are taken over there! Thank you again @Mossaka

**Pull request closed without merging**